crates/bevy_ecs/src/observer/mod.rs

```diff
@@ -192,13 +200,13 @@ impl ObserverDescriptor {
 #[derive(Debug)]
 pub struct ObserverTrigger {
     /// The [`Entity`] of the observer handling the trigger.
-    pub observer: Entity,
-
-    /// The [`ComponentId`] the trigger targeted.
-    pub event_type: ComponentId,
-
+    observer: Entity,
+    /// The [`Event`] the trigger targeted.
+    event_type: ComponentId,
+    /// The [`ComponentId`]s the trigger targeted.
+    components: SmallVec<[ComponentId; 1]>,
```

Comment:
I believe if we enable the [`union` feature](https://docs.rs/smallvec/latest/smallvec/#union) for `smallvec` (if it isn't already!), you can replace `[ComponentId; 1]` with `[ComponentId; 2]` within the same footprint due to alignment.